File: 3740-minimum-distance-between-three-equal-elements-i/3740-minimum-distance-between-three-equal-elements-i.py
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n = len(nums)
        res = inf
        for i in range(n - 2):
            for j in range(i + 1, n - 1):
                for k in range(j + 1, n):
                    if nums[i] == nums[j] == nums[k]:
                        res = min(res, 2 * (k - i))
        return res if res != inf else -1

File: 3740-minimum-distance-between-three-equal-elements-i/3740-minimum-distance-between-three-equal-elements-i.java
class Solution {
    public int minimumDistance(int[] nums) {
        int res = Integer.MAX_VALUE;
        int n = nums.length;
        for (int i = 0; i < n - 2; i++) {
            for (int j = i + 1; j < n - 1; j ++) {
                if (nums[i] != nums[j]) continue;
                for (int k = j + 1; k < n; k++) {
                    if (nums[j] != nums[k]) continue;
                    res = Math.min(res, 2 * (k - i));
                }
            }
        }
        return res == Integer.MAX_VALUE ? -1 : res;
    }
}